contracts/CATFactory.sol

```diff
@@ -70,6 +68,6 @@ contract CATFactory is Ownable {
      * @return The total number of CATs.
      */
     function totalCATs() public view returns (uint256) {
-        return _tokenIds.current();
+        return _nextTokenId - 1; // Subtract 1 because the counter starts from 1
```

Comment:
The subtraction by 1 here is incorrect.

Reply:
At the beginning, there are 0 CATs deployed and `_nextTokenId` is 0... So, the correct number of CATs is equal to `_nextTokenId`.

Reply:
Subtracting 1 would cause underflow.